docs/developer/architecture/ci.rst

```diff
@@ -0,0 +1,122 @@
+CI
+==
+
+Pre-merge
+---------
+
+What should be tested in pre-merge on every ``user/*``, ``feature/*``,
+``improvement/*``, ``bugfix/*``, ``w/*`` branches. The pre-merge test should
+not long too much time (less than 40 minutes is great) so we can't
+test everything in pre-merge, we should only test building of the product
+and check that product still usable.
```

Comment:
> Related: what about other targets, e.g. failure rates? And how do we plan on tracking these goals? Or even enforce them?

?? What do you talk about ?

Reply:
You said one target is `<40mn runs`. What about others? We're trying to write down a set of "requirements" before drafting a "design".

Reply:
> others ?

Goals are now define at the very beginning of the document, here just say that because of `having day-to-day development and PRs merged faster` goal we need to reduce pre-merge